src/main/java/edu/kit/ipd/crowdcontrol/objectservice/database/operations/NotificationOperations.java

```diff
@@ -226,17 +200,58 @@ public Notification updateNotification(int id, Notification notification) {
             });
         }
 
-        return this.getNotification(record.getIdNotification()).orElseThrow(NotFoundException::new);
+        return this.getNotification(notificationRecord.getIdNotification()).orElseThrow(NotFoundException::new);
     }
 
     /**
-     * runs a sql-query in read-only mode
+     * Runs a sql-query in read-only mode
      *
-     * @param sql the seq to execute in read-only mode
+     * @param sql the sql to execute in read-only mode
+     * @return the results of the query as a stream
+     */
+    public Stream<Record> runReadOnlySQL(String sql) {
+        return readOnlyCreate.fetchStream(sql);
+    }
+
+
+    /**
+     * This method will check if the passed NotificationTokenRecords are already stored in the database.
+     * It does so by comparing the resultId of each record for the specified notification.
+     * If there are new NotificationTokenRecords, they will be stored to database.
      *
-     * @return the Result of the query
+     * @param newTokenRecords the new records from a current fetch query mapped to the NOTIFICATION_TOKEN.RESULT_ID
+     * @param notificationId  the id of the notification
+     * @return a list of newly added NotificationTokenRecords, that were not stored in database before
      */
-    public Result<Record> runReadOnlySQL(String sql) {
-        return readOnlyCreate.fetch(sql);
+    public List<NotificationTokenRecord> diffTokenRecords(Map<Integer, NotificationTokenRecord> newTokenRecords, int notificationId) {
+        if (newTokenRecords != null && !newTokenRecords.isEmpty()) {
+            List<Query> deletionQueries = new ArrayList<>();
+
+            try (Stream<NotificationTokenRecord> recordStream = create.selectFrom(NOTIFICATION_TOKEN)
+                    .where(NOTIFICATION_TOKEN.NOTIFICATION.eq(notificationId))
+                    .fetchLazy().stream()) {
```

Comment:
.fetchLazy().stream() is the same as calling directly .stream(), but i think .stream() closes correctly